config/scootconfig/workers_modules.go

```diff
@@ -69,7 +70,13 @@ type WorkersLocalConfig struct {
 }
 
 func (c *WorkersLocalConfig) Install(bag *ice.MagicBag) {
-	bag.Put(func() worker.WorkerFactory {
-		return workers.MakeInmemoryWorker
+	bag.Put(func(tmp *temp.TempDir) worker.WorkerFactory {
+		return InmemoryWorkerFactory(tmp)
 	})
 }
+
+func InmemoryWorkerFactory(tmp *temp.TempDir) worker.WorkerFactory {
```

Comment:
I thought about it, and while it would've been easier, it's not needed by thrift worker and I think it's cleaner to curry per-type vars in this case.